common/src/main/java/org/fiware/tmforum/common/repository/TmForumRepository.java

```diff
@@ -26,6 +28,36 @@ public TmForumRepository(GeneralProperties generalProperties, EntitiesApiClient
     }
 
     public <T> Mono<T> get(URI id, Class<T> entityClass) {
+
+        String idString = id.toString();
+        String[] parts = idString.split(":", 4);
+
+
+        if (parts.length < 3) {
+            log.warn("Invalid NGSI-LD ID format, expected at least 3 parts: {}", idString);
+            return Mono.empty();
+        }
+
+        String requestedType = parts[2];
+
+        // Extract entity type from MappingEnabled
+        MappingEnabled mappingAnnotation = entityClass.getAnnotation(MappingEnabled.class);
+
+        // prevent NullPointerException
+        if (mappingAnnotation == null) {
+            log.warn("Class {} missing @MappingEnabled annotation", entityClass.getName());
+            return retrieveEntityById(id)
```

Comment:
I think you can throw an IlleaglArgument here directly, since the fromEntity method requires the annotation and will throw it downstream anyways